pandas/tests/tslibs/test_timedeltas.py

```diff
@@ -34,3 +34,41 @@ def test_huge_nanoseconds_overflow():
     # GH 32402
     assert delta_to_nanoseconds(Timedelta(1e10)) == 1e10
     assert delta_to_nanoseconds(Timedelta(nanoseconds=1e10)) == 1e10
+
+
+class TestComputingDeltasWithTimezoneDifferences:
+    def test_fails_to_subtract_when_one_date_is_naive(self):
+        time1 = Timestamp("2020-10-28 22:08:00-04:00")
+        time2 = Timestamp("2020-10-28 22:08:00")
+
+        msg = (
+            "Cannot compute time difference between "
+            "timezone-aware and naive datetime values."
+        )
+        with pytest.raises(TypeError, match=msg):
+            time1 - time2
+
+        with pytest.raises(TypeError, match=msg):
+            time2 - time1
+
+    def test_subtracts_dates_of_different_timezones(self):
```

Comment:
use `pytest.mark.parametrize` to write this and the test below as a single function. Parametrize on the two timestamps and the expected Timedelta

Reply:
I don't really like this idea, i think each test covers a different piece of functionality:

1. can i subtract when the timezone offsets are not the same?
2. can i subtract when the type of each datetime is not the same?

to me, using parametrization would hide the intent of each test -- you have to read and comprehend 4 date-times and 2 time-deltas in order to understand what is different about the two cases being tested. this is explicit, you can read the test function name and right away you know why the test exists